### PR TITLE
feat: make voting work again

### PR DESCRIPTION
**Pull request by ljkbennett**

### CODE REVIEW

File: apps/juicebox_web/assets/js/videos/video-placeholder.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+.video-name {
+  padding: 10px;
+}
+
+.video-voting {
+  display: flex;
+  flex-flow: column;
+  justify-content: space-between;
+  height: 100%;
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/videos/video-placeholder.scss:15]: Properties should be ordered display, width, height

[Read more about it here.](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#propertysortorder)

File: apps/juicebox_web/assets/js/videos/video.js [hunk 2]
```diff
@@ -40,6 +41,8 @@ class Video extends Component {
   }
 
   render() {
+    console.log(this.props);
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/videos/video.js]: Unexpected console statement.

[Read more about it here.](http://eslint.org/docs/rules/no-console)

File: apps/juicebox_stream/lib/juicebox_stream/stream/server.ex [hunk 3]
```diff
@@ -138,8 +145,16 @@ defmodule JuiceboxStream.Stream.Server do
     {:reply, {:ok, id}, state}
   end
 
-  def handle_cast({:vote, track_id}, state) do
-    {:noreply, Control.vote(state, track_id)}
+  def handle_call({:vote_up, track_id}, _from, state) do
+    new_state = Control.vote_up(state, track_id)
+    broadcast(state.id, @actions[:QUEUE_UPDATED], %{ videos: new_state.queue })
```
Comment (ljkbennett) [inline apps/juicebox_stream/lib/juicebox_stream/stream/server.ex:150]: There is no whitespace around parentheses/brackets most of the time, but here there is.

File: apps/juicebox_stream/lib/juicebox_stream/stream/server.ex [hunk 4]
```diff
@@ -138,8 +145,16 @@ defmodule JuiceboxStream.Stream.Server do
...
+    {:reply, {:ok, new_state}, new_state}
+  end
+
+  def handle_call({:vote_down, track_id}, _from, state) do
+    new_state = Control.vote_down(state, track_id)
+    broadcast(state.id, @actions[:QUEUE_UPDATED], %{ videos: new_state.queue })
```
Comment (ljkbennett) [inline apps/juicebox_stream/lib/juicebox_stream/stream/server.ex:156]: There is no whitespace around parentheses/brackets most of the time, but here there is.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

File: apps/juicebox_web/assets/js/videos/video-page.js [hunk 5]
```diff
@@ -3,28 +3,43 @@ import { connect } from 'react-redux';
...
+      <div key="video-page" className={ styles['video-page'] }>
+        <Search streamId={ streamId } />
+        <div key="main-col" className={ styles['main-column'] }>
           <Video video={ playing }
                  playingStartTime={ playingStartTime }
                  expanded={ expanded }
-                 toggleExpanded={ toggleExpanded } />
-          <Reactions streamId={this.props.streamId} />
+                 toggleExpanded={ toggleExpanded }
+                 key="video-wrapper"
```
Comment (OCannings) [inline apps/juicebox_web/assets/js/videos/video-page.js]: do we need these keys?

File: apps/juicebox_web/assets/js/videos/video.js [hunk 6]
```diff
@@ -13,7 +13,8 @@ class Video extends Component {
                 height="157"
                 src={`https://www.youtube.com/embed/${this.props.video.video_id}?autoplay=1&start=${this.props.playingStartTime}`}
                 frameBorder="0"
-                allowFullScreen></iframe>
+                allowFullScreen
+                key="youtube"></iframe>
```
Comment (OCannings) [inline apps/juicebox_web/assets/js/videos/video.js]: another key

File: apps/juicebox_stream/lib/juicebox_stream/stream/control.ex [hunk 7]
```diff
@@ -22,12 +22,14 @@ defmodule JuiceboxStream.Stream.Control do
   end
 
   def add_track(%{queue: queue} = state, track) do
-    new_queue = queue ++ [track]
+    # TODO: use Track struct
```
Comment (ljkbennett) [inline apps/juicebox_stream/lib/juicebox_stream/stream/control.ex:25]: TODO found

File: apps/juicebox_stream/test/juicebox_stream/stream/control_test.exs [hunk 8]
```diff
@@ -11,9 +11,9 @@ defmodule JuiceboxStream.Stream.ControlTests do
         timer: nil
       },
       videos: {
-        %{ track_id: 1, votes: 0 },
-        %{ track_id: 2, votes: 0 },
-        %{ track_id: 3, votes: 0 }
+        %{ video: %{ video_id: 1 }, votes: 0 },
```
Comment (ljkbennett) [inline apps/juicebox_stream/test/juicebox_stream/stream/control_test.exs:14]: There is no whitespace around parentheses/brackets most of the time, but here there is.

File: apps/juicebox_stream/test/juicebox_stream/stream/control_test.exs [hunk 9]
```diff
@@ -11,9 +11,9 @@ defmodule JuiceboxStream.Stream.ControlTests do
...
+        %{ video: %{ video_id: 2 }, votes: 0 },
```
Comment (ljkbennett) [inline apps/juicebox_stream/test/juicebox_stream/stream/control_test.exs:15]: There is no whitespace around parentheses/brackets most of the time, but here there is.

File: apps/juicebox_stream/test/juicebox_stream/stream/control_test.exs [hunk 10]
```diff
@@ -11,9 +11,9 @@ defmodule JuiceboxStream.Stream.ControlTests do
...
+        %{ video: %{ video_id: 3 }, votes: 0 }
```
Comment (ljkbennett) [inline apps/juicebox_stream/test/juicebox_stream/stream/control_test.exs:16]: There is no whitespace around parentheses/brackets most of the time, but here there is.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).
* 3 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).
* 3 fixed issues! 🎉

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/voting...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).
* 3 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

File: apps/juicebox_web/assets/js/videos/video-page.js [hunk 11]
```diff
@@ -2,28 +2,46 @@ import React, { Component, PropTypes } from 'react';
...
+      playingStartTime,
+      queue,
+      streamId,
+      toggleExpanded,
+      votes,
+      voteDown,
+      voteUp,
+    } = this.props;
+
+    console.log('votes', votes);
```
Comment (ljkbennett) [inline apps/juicebox_web/assets/js/videos/video-page.js]: Unexpected console statement.

[Read more about it here.](http://eslint.org/docs/rules/no-console)

**Comment by littleowllabs** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 6 possible new issues (including those that may have been commented here).
* 3 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

**Review by OCannings — APPROVED** [review]

**Comment by ljkbennett** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).
* 3 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/80.

**Pull request merged**